packages/text-input-react/src/BaseInputField.tsx

```diff
@@ -1,53 +1,54 @@
-import React, { FocusEventHandler, ChangeEventHandler, CSSProperties } from "react";
-import { LabelVariant } from "@fremtind/jkl-core";
+import React, { ChangeEventHandler, FocusEventHandler, CSSProperties, forwardRef } from "react";
 
-export interface BaseInputProps {
+export interface BaseProps {
+    id?: string;
+    className?: string;
+    maxLength?: number;
+    width?: string;
+    value?: string;
     onChange?: ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement>;
     onBlur?: FocusEventHandler<HTMLInputElement | HTMLTextAreaElement>;
     onFocus?: FocusEventHandler<HTMLInputElement | HTMLTextAreaElement>;
-    value?: string;
-    className?: string;
-    id?: string;
-    required?: boolean;
     placeholder?: string;
-    variant?: LabelVariant;
-    forceCompact?: boolean;
     readOnly?: boolean;
-    maxLength?: number;
-    width?: string;
     autoComplete?: string;
+    required?: boolean;
+    type?: "text" | "number" | "tel" | "password" | "email" | "year";
 }
 
-interface Props extends BaseInputProps {
-    type?: "text" | "number" | "tel" | "password" | "email" | "year";
+interface Props extends BaseProps {
+    describedBy?: string;
+    style?: CSSProperties;
     invalid?: boolean;
-    ariaDescribedby?: string;
 }
 
 function getWidthAsStyle(width?: string, maxLength?: number): CSSProperties | undefined {
     if (width) {
         return { width }; // prioritize width prop
     }
 
-    if (maxLength && maxLength < 15) {
-        return { width: `${maxLength + 3}ch` }; // else use maxLength if not large
+    if (maxLength) {
+        // adapt to maxLength, but capped at 40ch
+        const length = Math.min(maxLength, 40);
```

Comment:
Foreslår at vi logger en feil i dev om man har sendt inn et større tall. Ikke helt obvious hvorfor jeg ikke kan sende inn 100 hvis jeg vil.

Reply:
Man kan fortsatt sende inn `maxLength` over 40 for å begrense lendgen på _innholdet_, men lengden på _feltet_ vil ikke bli større enn om det var 40 `maxLength`. Det er sikkert greit med en melding uansett for å forlkare hva som skjer.